chalk-parse/src/parser.lalrpop

```diff
@@ -102,6 +103,21 @@ FnArgs: Vec<Ty> = {
     <Comma<FnArg>>
 };
 
+ClosureDefn: ClosureDefn = {
+    "closure" <n:Id> "(" <s:ClosureSelf> ")" ";" => ClosureDefn {
+        name: n,
+        kind: s,
+        argument_types: vec![],
+        return_type: Ty::Tuple { types: Vec::new() },
+    }
+}
+
+ClosureSelf: ClosureKind = {
+    "self" => ClosureKind::FnOnce,
+    "mutself" => ClosureKind::FnMut,
+    "refself" => ClosureKind::Fn,
```

Comment:
@nikomatsakis So question about lalrpop: If I use `self`, `&mut self`, and `&self` here instead, I get the following error in lowering:
`lowering error: parse error: UnrecognizedToken { token: (35, Token(7, "&"), 36), expected: ["\"&mut self\"", "\"&self\"", "\"self\""] }`
What's the "correct" way to do this?

Reply:
@jackh726 you probably want `"&" "self"` or something like that, i.e., two distinct tokens.